commit e07eff84451a284614f9c6c7d0daaa045308ac26
Author: Robin Dong <hao.bigrat@gmail.com>
Date:   Mon Mar 12 13:44:58 2012 +0800

    UBUNTU: ubuntu: overlayfs -- overlayfs: fix possible leak in ovl_new_inode
    
    After allocating a new inode, if the mode of inode is incorrect, we should
    release it by iput().
    
    Signed-off-by: Robin Dong <sanbai@taobao.com>
    Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>

diff --git a/fs/overlayfs/inode.c b/fs/overlayfs/inode.c
index c220ea7..86bf663 100644
--- a/fs/overlayfs/inode.c
+++ b/fs/overlayfs/inode.c
@@ -376,6 +376,7 @@ struct inode *ovl_new_inode(struct super_block *sb, umode_t mode,
 
 	default:
 		WARN(1, "illegal file type: %i\n", mode);
+		iput(inode);
 		inode = NULL;
 	}
 
